content/800-guides/230-betterauth-nextjs.mdx

```diff
@@ -0,0 +1,996 @@
+  const [error, setError] = useState<string | null>(null);
+  //add-end
+
+  return (
+    <main className="max-w-md mx-auto p-6 space-y-4 text-white">
+      <h1 className="text-2xl font-bold">Sign Up</h1>
+    </main>
+  );
+}
+```
+
+Now, import the `signUp` function from your Better-Auth client and add the `handleSubmit` function. This function is triggered on form submission and calls the `signUp.email` method provided by Better-Auth, passing the user's name, email, and password.
+
+```tsx file=src/app/sign-up/page.tsx
+"use client";
+
+import { useState } from "react";
+import { useRouter } from "next/navigation";
+//add-next-lin
+import { signUp } from "@/lib/auth-client";
+
+export default function SignUpPage() {
+  const router = useRouter();
+  const [error, setError] = useState<string | null>(null);
+
+  //add-start
+  async function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    setError(null);
+
+    const formData = new FormData(e.currentTarget);
+
+    const res = await signUp.email({
+      name: formData.get("name") as string,
+      email: formData.get("email") as string,
+      password: formData.get("password") as string,
+    });
+
+    if (res.error) {
+      setError(res.error.message || "Something went wrong.");
+    }
+  }
+  //add-end
+
+  return (
+    <main className="max-w-md mx-auto p-6 space-y-4 text-white">
+      <h1 className="text-2xl font-bold">Sign Up</h1>
+    </main>
+  );
+}
+```
+
+To inform the user of any issues, add an element that conditionally renders when the `error` state is not null.
+
+```tsx file=src/app/sign-up/page.tsx
+"use client";
+
+import { useState } from "react";
+import { useRouter } from "next/navigation";
+import { signUp } from "@/lib/auth-client";
+
+export default function SignUpPage() {
+  const router = useRouter();
+  const [error, setError] = useState<string | null>(null);
+
+  async function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    setError(null);
+
+    const formData = new FormData(e.currentTarget);
+
+    const res = await signUp.email({
+      name: formData.get("name") as string,
+      email: formData.get("email") as string,
+      password: formData.get("password") as string,
+    });
+
+    if (res.error) {
+      setError(res.error.message || "Something went wrong.");
+    }
+  }
+
+  return (
+    <main className="max-w-md mx-auto p-6 space-y-4 text-white">
+      <h1 className="text-2xl font-bold">Sign Up</h1>
+
+      //add-start
+      {error && <p className="text-red-500">{error}</p>}
+      //add-end
+    </main>
+  );
+}
+```
+
+Finally, add the HTML form with input fields for the user's name, email, and password, and a submit button.
+
+```tsx file=src/app/sign-up/page.tsx
+"use client";
+
+import { useState } from "react";
+import { useRouter } from "next/navigation";
+import { signUp } from "@/lib/auth-client";
+
+export default function SignUpPage() {
+  const router = useRouter();
```

Comment:
I get an ESLint error here:
"'router' is declared but its value is never read."

We can remove this if not needed.